app/services/pickup_rate/monitor.py

```
@@ -0,0 +1,348 @@
+            reseller_id=config.reseller_id,
+            merchant_id=config.merchant_id,
+        )
+
+        if rates is None:
+            logger.error(
+                f"PickupRateMonitor: DB query failed for scope='{config.scope}', "
+                "skipping alert cycle"
+            )
+            return
+
+        # Skip when there is no data – avoids false alerts on idle systems
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)